src/main/java/org/passay/CharacterRule.java

```diff
@@ -85,14 +85,14 @@ public RuleResult validate(final PasswordData passwordData)
   {
     final String matchingChars = PasswordUtils.getMatchingCharacters(
       String.valueOf(characterData.getCharacters()),
-      passwordData.getPassword());
-    if (matchingChars.length() >= numCharacters) {
-      return new RuleResult(true);
+      passwordData.getPassword(),
+      numCharacters);
+    if (matchingChars.length() < numCharacters) {
+      return new RuleResult(
+        false,
+        new RuleResultDetail(characterData.getErrorCode(), createRuleResultDetailParameters(matchingChars)));
     } else {
```

Comment:
No else block is required here since all it does is return the other case. Simply return it.